source/Mission.cpp

```diff
@@ -815,19 +815,19 @@ bool Mission::CanOffer(const PlayerInfo &player, const shared_ptr<Ship> &boardin
 		return false;
 
 	auto it = actions.find(OFFER);
-	if(it != actions.end() && !it->second.CanBeDone(player, boardingShip))
+	if(it != actions.end() && !it->second.CanBeDone(player, IsFailed(player), boardingShip))
 		return false;
 
 	it = actions.find(ACCEPT);
-	if(it != actions.end() && !it->second.CanBeDone(player, boardingShip))
+	if(it != actions.end() && !it->second.CanBeDone(player, IsFailed(player), boardingShip))
 		return false;
 
 	it = actions.find(DECLINE);
-	if(it != actions.end() && !it->second.CanBeDone(player, boardingShip))
+	if(it != actions.end() && !it->second.CanBeDone(player, IsFailed(player), boardingShip))
 		return false;
 
 	it = actions.find(DEFER);
-	if(it != actions.end() && !it->second.CanBeDone(player, boardingShip))
+	if(it != actions.end() && !it->second.CanBeDone(player, IsFailed(player), boardingShip))
```

Comment:
Why not evaluate that function once? I don't see how the result would change inside this context